apps/zk-explorer/components/RadioCard.tsx

```diff
@@ -0,0 +1,79 @@
+import { InfoCircledIcon } from '@radix-ui/react-icons';
+import { IconProps } from '@radix-ui/react-icons/dist/types';
+import { Typography } from '@webb-tools/webb-ui-components';
+import { ComponentType, FC, useCallback } from 'react';
+import { twMerge } from 'tailwind-merge';
+
+export type RadioCardProps = {
+  id: string;
+  title: string;
+  tooltip?: string;
+  description?: string;
+  Icon?: ComponentType<IconProps>;
+  selectedRadioItem: string | null;
+  setSelectedRadioItem: (id: string) => void;
+};
+
+export const RadioCard: FC<RadioCardProps> = ({
+  id,
+  setSelectedRadioItem,
+  selectedRadioItem,
+  title,
+  tooltip,
+  description,
+  Icon,
+}) => {
+  const isChecked = selectedRadioItem === id;
+
+  const handleOnChange = useCallback(() => {
+    {
+      if (selectedRadioItem === id) {
+        return;
+      }
+
+      setSelectedRadioItem(id);
+    }
+  }, [id, selectedRadioItem, setSelectedRadioItem]);
+
+  const isCheckedClass = isChecked
```

Comment:
Add useMemo

Reply:
The comparison is a constant time operation, so adding useMemo here might introduce overhead.

I did something a bit different however: Added `useMemo` for the `twMerge` calls, which is likely O(n). Going to start doing that everywhere I use `twMerge` that involves an input prop of unknown length.